hi-ml/testhiml/testhiml/test_run_ml.py

```diff
@@ -314,4 +315,62 @@ def test_runner_end_to_end() -> None:
         runner.setup()
         runner.init_training()
         runner.run_training()
-        assert True
+
+
+@pytest.mark.parametrize("log_from_vm", [True, False])
+def test_log_on_vm(log_from_vm: bool) -> None:
+    """Test if the AzureML logger is called when the experiment is run outside AzureML."""
+    experiment_config = ExperimentConfig(model="HelloWorld")
+    container = HelloWorld()
+    container.max_epochs = 1
+    # Mimic an experiment name given on the command line.
+    experiment_name = "unittest"
+    container.experiment = experiment_name
+    # The tag is used to identify the run, similar to the behaviour when submitting a run to AzureML.
+    tag = f"test_log_on_vm [{log_from_vm}]"
+    container.tag = tag
+    container.log_from_vm = log_from_vm
+    runner = MLRunner(experiment_config=experiment_config, container=container)
+    # When logging to AzureML, need to provide the unit test AML workspace.
+    # When not logging to AzureML, no workspace (and no authentication) should be needed.
+    if log_from_vm:
+        with patch("health_azure.utils.get_workspace", return_value=DEFAULT_WORKSPACE.workspace):
+            runner.run()
+    else:
+        runner.run()
+    # The PL trainer object is created in the init_training method.
+    # Check that the AzureML logger is set up correctly.
+    assert runner.trainer is not None
+    assert runner.trainer.loggers is not None
+    assert len(runner.trainer.loggers) > 1
+    logger = runner.trainer.loggers[1]
+    assert isinstance(logger, AzureMLLogger)
+    if log_from_vm:
+        assert logger.run is not None
+        # Check that all user supplied data (experiment and display name) are respected.
+        assert logger.run.experiment is not None
+        assert logger.run.experiment.name == experiment_name
+        assert logger.run.display_name == tag
+        # Both trainig and inference metrics must be logged in the same Run object.
+        metrics = logger.run.get_metrics()
+        assert "test_mse" in metrics
+        assert "loss" in metrics
+        # The run must have been correctly marked as completed. However, I did not find a way to check this.
+        # When we get here, the run is still returning "Running" as the status, even though it is completed in the UI.
```

Comment:
Maybe you need to call `run.wait_for_completion` ?

Reply:
(facepalm) Thanks Kenzi, that did the trick :-)